Add "how it works" section in readme (#271)
polyvariant/pitgull: README.md

# Pitgull

[![License](https://img.shields.io/:license-Apache%202-green.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
![Continuous Integration](https://github.com/pitgull/pitgull/workflows/Continuous%20Integration/badge.svg)
[![Powered by cats](https://img.shields.io/badge/powered%20by-cats-blue.svg)](https://github.com/typelevel/cats)
![Gluten free](https://img.shields.io/badge/gluten-free-orange.svg)

## How it works

> The core idea behind the project is very simple - create a tool for automatically applying merge requests created by [Scala Steward](https://github.com/scala-steward-org/scala-steward) on any [Gitlab](https://gitlab.com/) instance.

Reviewer: That was the core idea, but the app's goal is to be orthogonal from the bots making the PRs

Author: Fair point, added a sentence about this.

> Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. Once it's merged, Gitlab would trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.

Reviewer: ```suggestion
Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all the qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.
Once the MR merged, Gitlab would trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.
```

# Pitgull

[![License](https://img.shields.io/:license-Apache%202-green.svg)](http://www.apache.org/licenses/LICENSE-2.0.txt)
![Continuous Integration](https://github.com/pitgull/pitgull/workflows/Continuous%20Integration/badge.svg)
[![Powered by cats](https://img.shields.io/badge/powered%20by-cats-blue.svg)](https://github.com/typelevel/cats)
![Gluten free](https://img.shields.io/badge/gluten-free-orange.svg)

## How it works

The core idea behind the project is very simple - create a tool for automatically applying merge requests created by [Scala Steward](https://github.com/scala-steward-org/scala-steward) on any [Gitlab](https://gitlab.com/) instance. Currently Pitgull is ortogonal from the bots, the merge requests are qualified by generic rules.

> Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all the qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.

Reviewer 2: ```suggestion
Pitgull works as a web service, listening for webhook events from Gitlab instance. Upon receiving a webhook, it reads related project's merge requests, finds one best candidate and tries to merge it. If all qualifying merge requests need to be updated, an attempt to rebase them onto the target branch will be made.
```

> Once the MR merged, Gitlab would trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.

Reviewer 2: ```suggestion
Once an MR is merged, Gitlab will trigger another webhook, creating a loop until Pitgull finds no qualifying merge request.
```
